src/owner.rs

```diff
@@ -0,0 +1,357 @@
+    github_id: i32,
+    /// Unique table id
+    cargo_id: i32,
+    /// "github:org:team"
+    /// An opaque unique ID, that was at one point parsed out to query Github.
+    /// We only query membership with github using the github_id, though.
+    name: String,
+}
+
+#[derive(RustcDecodable, RustcEncodable)]
+pub struct EncodableOwner {
+    pub id: i32,
+    // Login is deprecated in favour of name, but needs to be printed for back-compat
+    pub login: String,
+    pub kind: String,
+    pub email: Option<String>,
+    pub name: Option<String>,
+    pub avatar: Option<String>,
+}
+
+/// Access rights to the crate (publishing and ownership management)
+/// NOTE: The order of these variants matters!
+#[derive(PartialEq, Eq, PartialOrd, Ord)]
+pub enum Rights {
+    None,
+    Publish,
+    Full,
+}
+
+impl Team {
+    /// Just gets the Team from the database by name.
+    pub fn find_by_name(conn: &Connection, name: &str) -> CargoResult<Self> {
+        let stmt = try!(conn.prepare("SELECT * FROM teams
+                                      WHERE name = $1"));
+        let rows = try!(stmt.query(&[&name]));
+        let row = try!(rows.iter().next().chain_error(|| {
+            NotFound
+        }));
+        Ok(Model::from_row(&row))
+    }
+
+    /// Tries to create the Team in the DB (assumes a `:` has already been found).
+    pub fn create(conn: &Connection, name: &str, req_user: &User) -> CargoResult<Self> {
+        // must look like system:xxxxxxx
+        let mut chunks = name.split(":");
+        match chunks.next().unwrap() {
+            // github:rust-lang:owners
+            "github" => {
+                // Ok to unwrap since we know one ":" is contained
+                let org = try!(chunks.next().ok_or_else(||
+                    human("missing github org argument; format is github:org:team")
+                ));
+                let team = try!(chunks.next().ok_or_else(||
+                    human("missing github team argument; format is github:org:team")
+                ));
+                Team::create_github_team(conn, name, org, team, req_user)
+            }
+            _ => {
+                Err(human("unknown organization handler, only 'github:org:team' is supported"))
+            }
+        }
+    }
+
+    /// Tries to create a Github Team from scratch. Assumes `org` and `team` are
+    /// correctly parsed out of the full `name`. `name` is passed as a convenience
+    /// to avoid rebuilding it.
+    pub fn create_github_team(conn: &Connection, name: &str, org_name: &str, team_name: &str,
+                              req_user: &User) -> CargoResult<Self> {
+        // GET orgs/:org/teams
+        // check that `team` is the `slug` in results, and grab its `id`
+
+        // "sanitization"
+        fn whitelist(c: &char) -> bool {
+            match *c {
+                'a'...'z' | 'A'...'Z' | '0'...'9' | '-' | '_' => false,
+                _ => true
+            }
+        }
```

Comment:
That doesn't seem very semantically clear. We're trying to filter out to only the valid set of GH team slugs. (actually I think A...Z could be omitted)